modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -754,49 +754,35 @@ void ONNXImporter::parseOperatorSet()
     }
 }
 
-void ONNXImporter::handleQuantizedNode(LayerParams& layerParams,
+void ONNXImporter::setParamsDtype(LayerParams& layerParams,
                                        const opencv_onnx::NodeProto& node_proto)
 {
-    // Quantized nodes have output names ending with 'quantized'
-    std::string outName = node_proto.output(0);
-    int len = outName.length();
-    if (len <= 9)
-        return;
-
-    if (outName.substr(len - 9) == "quantized")
+    bool hasVariableInps = false;
+    int i = 0;
+    for (; i < node_proto.input_size(); ++i)
     {
-        outName = outName.substr(0, len - 9);
-        Mat scale, zeropoint;
-
-        if (constBlobs.find(outName + "scale") != constBlobs.end() &&
-            constBlobs.find(outName + "zero_point") != constBlobs.end())
-        {
-            scale = getBlob(outName + "scale");
-            zeropoint = getBlob(outName + "zero_point");
-        }
-        else
+        if (layer_id.find(node_proto.input(i)) != layer_id.end())
         {
-            std::string inpName = node_proto.input(0);
-            inpName = inpName.substr(0, inpName.length() - 9);
-            scale = getBlob(inpName + "scale");
-            zeropoint = getBlob(inpName + "zero_point");
-
-            for (int i = 0; i < node_proto.output_size(); i++)
-            {
-                std::string out = node_proto.output(i);
-                out = out.substr(0, out.length() - 9);
-                addConstant(out + "scale", scale);
-                addConstant(out + "zero_point", zeropoint);
-            }
+            hasVariableInps = true;
+            break;
         }
+    }
 
-        if (scale.total() != 1 || zeropoint.total() != 1)
-            CV_Error(Error::StsNotImplemented, "Per-channel scales/zeropoints are not supported");
+    if (hasVariableInps)
+    {
+        LayerInfo layerInfo = layer_id.find(node_proto.input(i))->second;
+        Ptr<Layer> preLayer = dstNet.getLayer(layerInfo.layerId);
+        auto layerType = preLayer->type;
+        int len = layerType.length();
 
-        layerParams.set("depth", CV_8S);
-        layerParams.set("scales", DictValue::arrayReal(scale.ptr<float>(), 1));
-        layerParams.set("zeropoints", DictValue::arrayInt(zeropoint.ptr<int8_t>(), 1));
+        if (layerType == "Quantize"
+            || (len > 3 && layerType.substr(len - 4) == "Int8"))
+        {
+            layerParams.set("depth", CV_8S);
+            return;
+        }
     }
+    layerParams.set("depth", CV_32F);
```

Comment:
This code wants to get the depth of the current layer based on the node's input. Do we have a better way to get `depth` directly from the layer?

Reply:
Welp, layers have constrains on types in the spec, but we don't have the information in the graph which types this layer supports.